updateLib.js

```diff
@@ -0,0 +1,177 @@
+  const text = await fetchText(url);
+  return JSON.parse(text);
+};
+
+// nodeVersions is sorted
+const getLatestNodeVersion = (nodeVersions, majorVersion) => nodeVersions
+  .find((version) => version.startsWith(`${majorVersion}.`));
+
+const getTemplate = (variant) => {
+  if (variant.startsWith('alpine')) {
+    return templates.alpine;
+  }
+
+  if (variant.endsWith('-slim')) {
+    return templates.debianSlim;
+  }
+
+  return templates.debian;
+};
+
+const getDockerfileMetadata = (nodeVersions, file) => {
+  const [nodeMajorVersion, variant] = path.dirname(file).split(path.sep).slice(-2);
+  const fileNodeVersion = getDockerfileNodeVersion(file);
+
+  return {
+    file,
+    variant,
+    fileNodeVersion,
+    nodeMajorVersion,
+    latestVersion: getLatestNodeVersion(nodeVersions, nodeMajorVersion),
+    template: getTemplate(variant),
+  };
+};
+
+const isDockerfileOutdated = ({ fileNodeVersion, latestVersion }) => fileNodeVersion
+  !== latestVersion;
+
+const fetchLatestNodeVersions = async () => {
+  const nodeDist = await fetchJson(releaseUrl);
+  return nodeDist.map(({ version }) => version.substring(1));
+};
+
+const findOutdated = async (updateAll) => {
+  const nodeVersions = await fetchLatestNodeVersions();
+
+  const dockerfileMetadatas = getAllDockerfiles(__dirname)
+    .map((file) => getDockerfileMetadata(nodeVersions, file));
+
+  return updateAll
+    ? dockerfileMetadatas
+    : dockerfileMetadatas.filter(isDockerfileOutdated);
+};
+
+const getKeys = (basename) => readFileSync(path.resolve(__dirname, 'keys', basename))
+  .toString().trim().split('\n');
+
+const readTemplate = (template) => readFileSync(
+  path.resolve(__dirname, templateFileMap[template]),
+).toString();
+
+const getBaseImage = ({ template, variant }) => {
+  const tag = template === templates.alpine
+    ? variant.replace(/alpine/, '')
+    : variant;
+
+  return `${templateRepoMap[template]}:${tag}`;
+};
+
+const formatKeys = (keys) => keys.map((key) => `$1${key} \\`).join('\n');
+
+const formatTemplate = (yarnVersion, nodeKeys, yarnKeys, muslChecksum, base, metadata) => {
+  const { latestVersion, template, nodeMajorVersion } = metadata;
+  const baseImage = getBaseImage(metadata);
+  const initialFormat = base.replace(/^FROM.+$/m, `FROM ${baseImage}`)
+    .replace(/^ENV NODE_VERSION .+$/m, `ENV NODE_VERSION ${latestVersion}`)
+    .replace(/^ENV YARN_VERSION .+$/m, `ENV YARN_VERSION ${yarnVersion}`)
+    .replace(/^(\s*)"\${NODE_KEYS\[@]}".*$/m, formatKeys(nodeKeys))
+    .replace(/^(\s*)"\${YARN_KEYS\[@]}".*$/m, formatKeys(yarnKeys));
+
+  if (template !== templates.alpine) {
+    return initialFormat;
+  }
+
+  const pythonVersion = parseInt(nodeMajorVersion, 10) < 14
+    ? 'python2'
+    : 'python3';
+
+  return initialFormat.replace(/\${PYTHON_VERSION}/m, pythonVersion)
+    .replace(/CHECKSUM=CHECKSUM_x64/m, `CHECKSUM="${muslChecksum}"`);
+};
+
+const fetchMuslChecksum = async (nodeVersion) => {
+  const checksums = await fetchText(
```

Comment:
If the checksum isn't available, an action using this should fail since `fetchText` will reject on non 2XX status codes. Is that sufficient? I could instead have it periodically poll that site but that would run into the action time limit of 6 hours. Any thoughts?